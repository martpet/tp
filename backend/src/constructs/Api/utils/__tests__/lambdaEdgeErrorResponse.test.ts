import {
  errorResponse,
  itCalls,
  itReturns,
  lambdaEdgeErrorResponse,
} from '~/constructs/Api/utils';

vi.mock('../errorResponse');

const args = ['dummyArg1', 'dummyArg2'] as unknown as Parameters<
  typeof lambdaEdgeErrorResponse
>;

vi.mocked(errorResponse).mockReturnValue({
  statusCode: 418,
  body: 'dummyErrorResponseBody',
});

describe('lambdaEdgeErrorResponse', () => {
  itCalls(errorResponse, lambdaEdgeErrorResponse, args);
  itReturns(lambdaEdgeErrorResponse, args);
});
