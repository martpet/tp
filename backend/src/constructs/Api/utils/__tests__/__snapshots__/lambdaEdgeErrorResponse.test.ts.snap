// Vitest Snapshot v1

exports[`lambdaEdgeErrorResponse > calls "errorResponse" with correct args 1`] = `
[
  [
    "dummyArg1",
    "dummyArg2",
  ],
]
`;

exports[`lambdaEdgeErrorResponse > returns a correct value 1`] = `
{
  "body": "dummyErrorResponseBody",
  "headers": {
    "content-type": [
      {
        "value": "application/json",
      },
    ],
  },
  "status": 418,
  "statusDescription": "I'm a teapot",
}
`;
